### PR TITLE
feat: add TypeScript template to create LWC command

### PR DESCRIPTION
**Pull request by jmsjtu**
### What does this PR do?

This PR adds a new template to the "SFDX: Create Lightning Web Component" command.

The tests are not passing as it first requires this [PR](https://github.com/forcedotcom/salesforcedx-templates/pull/590) to be merged and released.

This PR also introduces a setting to gate the feature behind a flag.

### What issues does this PR fix or reference?
@W-15696151@

### CODE REVIEW

File: packages/salesforcedx-vscode-lwc/package.nls.ja.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
   "lightning_lwc_preferences": "Salesforce Lightning Web Components",
   "lightning_lwc_preview_text": "SFDX: コンポーネントをローカルでプレビュー",
   "lightning_lwc_remember_device_description": "最後に使用したモバイルデバイスのターゲットを保存する",
+  "lightning_lwc_preview_typescript_support": "最後に使用したモバイルデバイスのターゲットを保存する",
```
Comment (jmsjtu) [inline packages/salesforcedx-vscode-lwc/package.nls.ja.json:9]: I will contact the IDEx doc writer to get an official description, this is a placeholder for now.

File: packages/salesforcedx-vscode-core/src/commands/templates/lightningGenerateLwc.ts [hunk 2]
```diff
@@ -34,32 +35,42 @@ export class LibraryLightningGenerateLwcExecutor extends LibraryBaseTemplateComm
   public telemetryName = 'lightning_generate_lwc';
   public metadataTypeName = LWC_TYPE;
   public templateType = TemplateType.LightningComponent;
+  private templateOptions: LightningComponentOptions | undefined;
```
Comment (jmsjtu) [inline packages/salesforcedx-vscode-core/src/commands/templates/lightningGenerateLwc.ts:38]: The `LightningComponentOption` requires this [PR](https://github.com/forcedotcom/salesforcedx-templates/pull/590) to pass compilation.

**Review by mingxuanzhangsfdx — COMMENTED** [review]
Thanks for the change!

File: packages/salesforcedx-vscode-lwc/package.nls.ja.json (hunk 1 above)
Comment (jmsjtu) [reply packages/salesforcedx-vscode-lwc/package.nls.ja.json]: Spoke with doc writers and this change is not needed.

**Review by mingxuanzhangsfdx — APPROVED** [review]
Looks good. The change will not effect the existing function because the setting flag is not explicit right now. https://github.com/forcedotcom/salesforcedx-vscode/actions/runs/10084577668/job/27883675234 e2e test runs succesfully.

**Pull request merged**